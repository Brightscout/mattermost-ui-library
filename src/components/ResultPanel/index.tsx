import React, {forwardRef} from 'react';

import {CheckIcon} from '@Components/Icons';

type ResultPanelProps = {
    iconClass?: string | null;
    header: string | null;
    className?: string;
    primaryBtn?: {
        text: string,
        onClick: (() => void) | null,
    }
    secondaryBtn?: {
        text: string,
        onClick: (() => void) | null,
    }
};

const ResultPanel = forwardRef<HTMLDivElement, ResultPanelProps>(({
    header,
    className = '',
    primaryBtn,
    secondaryBtn,
    iconClass,
}: ResultPanelProps, resultPanelRef): JSX.Element => (
    <div
        className={`modal__body modal-body d-flex align-items-center justify-content-center flex-column wizard__secondary-panel ${className}`}
        ref={resultPanelRef}
    >
        {iconClass ? (
            <i className={`fa result-panel-icon ${iconClass ?? 'fa-check-circle-o'}`}/>
        ) : (
            <CheckIcon className='result-panel-icon'/>
        )}
        <h2 className='font-16 margin-v-25 text-center text-capitalize'>{header}</h2>
        {primaryBtn?.onClick && (
            <button
                className='btn btn-primary'
                onClick={primaryBtn.onClick}
            >
                {primaryBtn.text}
            </button>
        )}
        {secondaryBtn?.onClick && (
            <button
                className='btn btn-link margin-top-12 padding-0'
                onClick={secondaryBtn.onClick}
            >
                {secondaryBtn.text}
            </button>
        )}
    </div>
));

export default ResultPanel;
