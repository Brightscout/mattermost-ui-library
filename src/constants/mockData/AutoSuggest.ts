// Mock data for "AutoSuggest" component tests
export const onInputValueChange = jest.fn();
export const onOptionClick = jest.fn();
export const suggestionConfig = {
    suggestions: [{
        label: 'Label 1',
        value: 'Value 1',
    }, {
        label: 'Label 2',
        value: 'Value 2',
    }, {
        label: 'Label 3',
        value: 'Value 3',
    }],
    renderValue: (suggestion: Record<string, string>) => suggestion.label,
};
