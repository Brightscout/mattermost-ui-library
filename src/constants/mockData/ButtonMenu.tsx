import React from 'react';

// Mock data for "ButtonMenu" component tests
export const Icon = <i className='icon icon-globe' />;
export const onBtnClick = jest.fn();
export const buttons = [{
    text: 'Button Text',
    icon: Icon,
    onClick: onBtnClick,
}, {
    text: 'Button Text 2',
    icon: Icon,
    onClick: onBtnClick,
}];
