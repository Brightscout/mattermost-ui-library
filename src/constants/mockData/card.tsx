import React from 'react';

// Mock data for "card" related component tests

// Mock data for "BaseCard"
export const BaseCardChildren = <div>{'Base card children'}</div>;

// Mock data for "SubscriptionCard"
export const editHandler = jest.fn();
export const deleteHandler = jest.fn();
export const subscriptionCardBody = {
    list: ['List Item 1', 'List Item 2', 'List Item 3'],
    labelValuePairs: [
        {label: 'Label 1', value: 'Value 1'},
        {label: 'Label 2', value: 'Value 2'},
        {label: 'Label 3', value: 'Value 3'},
        {label: 'Label 4', value: 'Value 4'},
    ],
};
// Mock Icons
export const EditIcon = <i className='fa fa-edit' />;
export const DeleteIcon = <i className='fa fa-delete' />;
// Mock channel data
export const channelData = {
    create_at: 19304124124,
    creator_id: 'klsioer239042',
    delete_at: 19304124124,
    display_name: 'Channel Display Name',
    extra_update_at: 19304124124,
    group_constrained: null,
    header: 'Channel Header',
    id: 'jw390r23iofs9032',
    last_post_at: 19304124124,
    last_root_post_at: 19304124124,
    name: 'Channel Name',
    policy_id: null,
    props: null,
    purpose: 'Channel Purpose',
    scheme_id: null,
    shared: null,
    team_display_name: 'Team Display Name',
    team_id: 'ior9823ijoiewro3w',
    team_name: 'Team Name',
    team_update_at: 19304124124,
    total_msg_count: 1,
    total_msg_count_root: 1,
    type: 'P',
    update_at: 19304124124,
};
export const subscriptionCardProps = {
    header: 'Subscription Card Header',
    label: 'Label',
    cardBody: subscriptionCardBody,
    description: 'Description',
    channel: channelData,
    editIcon: EditIcon,
    deleteIcon: DeleteIcon,
    onEdit: editHandler,
    onDelete: deleteHandler,
    className: 'subscription-card-custom-class',
};

// Mock data for "Card" component
export const cardBody = [{
    sectionHeading: 'Section Heading',
    data: [{
        icon: <i className='fa fa-edit' />,
        label: 'Label 1',
    }, {
        icon: <i className='fa fa-link' />,
        label: 'Label 2',
    }],
}];
export const cardProps = {
    cardHeader: 'Card Header',
    cardSubHeader: 'Card sub header',
    cardBody,
    onEdit: editHandler,
    onDelete: deleteHandler,
    editIcon: EditIcon,
    deleteIcon: DeleteIcon,
    className: 'card-custom-class',
};
