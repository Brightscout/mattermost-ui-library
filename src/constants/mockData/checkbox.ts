// Mock data for "Checkbox" component
export const checkboxLabel = 'Checkbox Label';
export const checkboxChangeHandler = jest.fn();
export const changeHandler = jest.fn();
