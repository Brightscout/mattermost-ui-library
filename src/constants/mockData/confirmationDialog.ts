
// Mock data for "ConfirmationDialog" component
export const hideHandler = jest.fn();
export const confirmHandler = jest.fn();
export const confirmationDialogProps = {
    title: 'Dialog Title',
    confirmationMsg: 'Dialog Message',
    show: true,
    onHide: hideHandler,
    loading: true,
    onConfirm: confirmHandler,
    className: 'confirmation-dialog-class',
    error: 'Dialog error',
};
