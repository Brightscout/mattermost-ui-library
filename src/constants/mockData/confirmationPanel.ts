// Mock data for "ConfirmationPanel" component
export const hideHandler = jest.fn();
export const confirmHandler = jest.fn();
export const confirmationPanelProps = {
    confirmationMsg: 'Dialog Message',
    onHide: hideHandler,
    onConfirm: confirmHandler,
    loading: true,
    className: 'confirmation-dialog-class',
    error: 'Dialog error',
};
