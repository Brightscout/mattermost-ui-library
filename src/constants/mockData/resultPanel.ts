// Mock data for "ResultPanel" component
export const primaryBtnHandler = jest.fn();
export const secondaryBtnHandler = jest.fn();
export const primaryBtnText = 'Primary Button';
export const secondaryBtnText = 'Secondary Button';
export const resultPanelProps = {
    iconClass: 'icon-class',
    header: 'Result Panel Header',
    className: 'result-panel-class',
    primaryBtn: {
        text: primaryBtnText,
        onClick: primaryBtnHandler,
    },
    secondaryBtn: {
        text: secondaryBtnText,
        onClick: secondaryBtnHandler,
    },
};
