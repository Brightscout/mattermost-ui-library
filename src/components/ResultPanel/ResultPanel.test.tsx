import React from 'react';
import {mount, ReactWrapper} from 'enzyme';

// Mock data
import {primaryBtnHandler, secondaryBtnHandler, resultPanelProps} from '@Constants/mockData/resultPanel';

import ResultPanel from '.';

describe('Result Panel', () => {
    let component: ReactWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

    beforeEach(() => {
        component = mount(<ResultPanel {...resultPanelProps}/>);
    });

    afterEach(() => {
        component.unmount();
        jest.clearAllMocks();
    });

    it('Should render correctly', () => {
        expect(component).toMatchSnapshot();
    });

    it('Should apply the passed "className" prop', () => {
        expect(component.hasClass(resultPanelProps.className)).toBeTruthy();
    });

    it('Should apply the class to the icon', () => {
        expect(component.find('.result-panel-icon').hasClass(resultPanelProps.iconClass)).toBeTruthy();
    });

    it('Should call the "onClick" handler when the primary button is clicked', () => {
        component.find('.btn-primary').simulate('click');
        expect(primaryBtnHandler).toHaveBeenCalledTimes(1);
    });

    it('Should call the "onClick" handler when the cancel button is clicked', () => {
        component.find('.btn-link').simulate('click');
        expect(secondaryBtnHandler).toHaveBeenCalledTimes(1);
    });
});

describe('Result Panel - skip rendering the buttons', () => {
    let component: ReactWrapper<any, Readonly<{}>, React.Component<{}, {}, any>>;

    beforeEach(() => {
        component = mount(<ResultPanel header={resultPanelProps.header}/>);
    });

    afterEach(() => {
        component.unmount();
    });

    it('Should not render the primary button when "primaryBtn" prop isn\'t passed', () => {
        expect(component.find('.btn-primary').exists()).toBeFalsy();
    });

    it('Should not render the secondary button when "secondaryBtn" prop isn\'t passed', () => {
        expect(component.find('.btn-link').exists()).toBeFalsy();
    });
});
