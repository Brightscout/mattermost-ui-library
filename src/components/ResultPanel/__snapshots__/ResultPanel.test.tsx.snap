// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result Panel Should render correctly 1`] = `
<ForwardRef
  className="result-panel-class"
  header="Result Panel Header"
  iconClass="icon-class"
  primaryBtn={
    Object {
      "onClick": [MockFunction],
      "text": "Primary Button",
    }
  }
  secondaryBtn={
    Object {
      "onClick": [MockFunction],
      "text": "Secondary Button",
    }
  }
>
  <div
    className="modal__body modal-body d-flex align-items-center justify-content-center flex-column wizard__secondary-panel result-panel-class"
  >
    <i
      className="fa result-panel-icon icon-class"
    />
    <h2
      className="font-16 margin-v-25 text-center text-capitalize"
    >
      Result Panel Header
    </h2>
    <button
      className="btn btn-primary"
      onClick={[MockFunction]}
    >
      Primary Button
    </button>
    <button
      className="btn btn-link margin-top-12 padding-0"
      onClick={[MockFunction]}
    >
      Secondary Button
    </button>
  </div>
</ForwardRef>
`;
